File: backend/src/main/java/de/neuefische/backend/model/Recipe.java
package de.neuefische.backend.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document(collection = "recipes")
public class Recipe {

    @Id
    private String id;
    private String title;
    private String image;
    private boolean vegetarian;
    private boolean vegan;
    private boolean glutenFree;
    private BigDecimal pricePerServing;
    private int readyInMinutes;
    private int servings;
    private String summary;
    private Ingredients[] extendedIngredients;
}

File: frontend/src/model/Recipe.ts
import {Ingredient} from "./Ingredient";
import {Instruction} from "./Instruction";
import {Equipment} from "./Equipment";

export type Recipe = {
    id: string;
    title: string;
    image?: string;
    vegetarian: boolean;
    vegan: boolean;
    glutenFree: boolean;
    readyInMinutes: number;
    servings: number;
    summary: string;
    extendedIngredients: Ingredient[];
    instructions?: string;
    analyzedInstructions: Instruction[];
    equipment?: Equipment[];
}
